File: api/routers/permutation.py
from fastapi import APIRouter, HTTPException
from pydantic import BaseModel
from ..services.permutation_engine import generate

router = APIRouter(prefix="/permutation", tags=["Permutation"])

class PermRequest(BaseModel):
    first_name: str
    last_name:  str
    domain:     str

@router.post("/generate")
async def generate_permutations(req: PermRequest):
    if not req.first_name or not req.last_name or not req.domain:
        raise HTTPException(422, "first_name, last_name, and domain are required.")
    patterns = generate(req.first_name, req.last_name, req.domain)
    return {"domain": req.domain, "patterns": patterns}

File: api/routers/permutation.js
import { Router } from "express";
import { generate } from "../services/permutationEngine.js";

export const permutationRouter = Router();

permutationRouter.post("/permutation/generate", (req, res) => {
  const { first_name, last_name, domain } = req.body;
  if (!first_name || !last_name || !domain) {
    return res.status(422).json({ error: "first_name, last_name, and domain are required." });
  }
  const patterns = generate(first_name, last_name, domain);
  res.json({ domain, patterns });
});
